### PR TITLE
Do not import disabled extensions

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/24250

The disabled extension should not be imported

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
- steps from issue

### CODE REVIEW

File: common/importer/chrome_importer_utils.cc [hunk 1]
```diff
@@ -157,6 +157,9 @@ std::vector<std::string> GetImportableListFromChromeExtensionsList(
     // installed by default.
     if (dict.FindBool("was_installed_by_default").value_or(true))
       continue;
+    // `"state": 0` means disabled state
+    if (!item.second.FindIntKey("state").value_or(false))
```
Comment (simonhong) [inline common/importer/chrome_importer_utils.cc:161]: Maybe `if (!dict.FindInt("state").value_or(0))`?

Comment (spylogsster) [reply common/importer/chrome_importer_utils.cc]: done, thanks!

**Review by simonhong — APPROVED** [review]
++

**Review by bsclifton — APPROVED** [review]
Change is nice and clean 😄 Works great - and thanks for adding the test

**Comment by kjozwiak** [comment]
Reproduced the original issue using `1.44.39 Chromium 104.0.5112.81` with the STR/Cases outlined via https://github.com/brave/brave-browser/issues/24250#issue-1317104395.

**`Importing from Chrome`**

**`Chrome`** | **`Before Import`** | **`After Import`**
---------------|----------------------|---------------------
image | BeforeReproduced | chromeAfterReproduced

**`Importing from Edge`**

**`Edge`** | **`Before Import`** | **`After Import`**
---------------|----------------------|---------------------
image | BeforeReproduced | edgeAfterReproduced

Verification **PASSED** on `Win 11 x64` using the following builds(s):

```
Brave | 1.44.44 Chromium: 104.0.5112.81 (Official Build) nightly (64-bit)
-- | --
Revision | 5b7b76419d50f583022568b6764b630f6ddc9208-refs/branch-heads/5112@{#1309}
OS | Windows 11 Version 21H2 (Build 22000.856)
```

**`Importing from Chrome`**

**`Chrome`** | **`Before Import`** | **`After Import`**
---------------|----------------------|---------------------
image | BeforeReproduced | chromeAfterFix

**`Importing from Edge`**

**`Edge`** | **`Before Import`** | **`After Import`**
---------------|----------------------|---------------------
image | BeforeReproduced | edgeAfterFix
 
* ensured that disabled extensions are not being imported via `brave://welcome`
* ensured that disabled extensions are not being imported via `brave://settings/importData`

**Pull request merged**